packages/ember-metal/lib/chains.ts

```diff
@@ -266,16 +258,14 @@ class ChainNode {
   }
 
   chain(key: string, tails: string[]): void {
-    let chains: { [key: string]: ChainNode | undefined };
-    if (this._chains === undefined) {
+    let chains = this._chains;
+    if (chains === undefined) {
       chains = this._chains = Object.create(null);
-    } else {
-      chains = this._chains;
     }
 
-    let node = chains[key];
+    let node = chains![key];
```

Comment:
the `!` shouldn't be needed because chains is already checked for undefined and assigned.

Reply:
linter is complaining if I do that